WebDevStudios/nextjs-wordpress-starter
# Feature/380 gf upload support

Closes #380

**Merge [BE PR #28](https://github.com/WebDevStudios/wds-headless-wordpress/pull/28) first**

### Description

Re-adds GF file upload field support:
- Re-enables file upload fields in GF forms
- Adds GraphQL schema type definitions for field values
- Adds custom Apollo body serializer to convert GF entry JSON to multipart form
- Adds `next-connect` and custom middleware to parse form data, allowing us to pass File objects from the client side through to our Next.js API route
- Converts File objects to standard JS objects before passing to WP side to ensure data is in correct format

### Screenshot

![Screen Shot 2021-05-20 at 11 13 22 AM](https://user-images.githubusercontent.com/36422618/119021260-753d1800-b95c-11eb-99ab-065747b8387b.png)

![Screen Shot 2021-05-20 at 11 13 36 AM](https://user-images.githubusercontent.com/36422618/119021258-74a48180-b95c-11eb-864f-d98525985047.png)

### Verification

(Not currently working on the preview env - might need to deploy dev WP env to test there)
1. Check out branch `feature/380-gf-upload-support` from this repo and `feature/380-gf-upload-support` from the [WP repo](https://github.com/WebDevStudios/wds-headless-wordpress/pull/28)
2. Run `npm i --legacy-peer-deps` on this repo and `composer install` on the WP repo
3. Start up dev env
4. Fill out form, including attachment field
5. Submit form - success message should appear
6. View [entries in WP admin](https://nextjsdevstart.wpengine.com/wp-admin/admin.php?page=gf_entries&id=3) - confirm submission recorded in full, including attachment
